### PR TITLE
allow join to be trusted again

### PR DESCRIPTION
**Pull request by hcli-work**
https://app.asana.com/0/1109150244034467/1199910849205333

We are getting a ton of errors in Canvas and students report bugs logging into Braven Network. With Brian's help, we saw that Canvas wasn't trusting the referrer when using the openid/url_script endpoint. The BASE_URL in production was already set to platform actually, so what seems to be the problem is that join is calling out to canvas, but canvas doesn't trust join.

This pull request just uses another ENV variable explicitly for the join server. I lightly refactored some of that code and grepped for what `BeyondZConfiguration.join_url` to see where it might make a different.

Test Plan: hit joinweb:30001/connect locally. it errored out before and now gets redirected.

### CODE REVIEW

File: docker-compose/.env [hunk 1]
```diff
@@ -12,14 +12,15 @@
 #AWS_ACCESS_KEY_ID=THIS_VALUE_SHOULD_HAVE_BEEN_PASSED_FROM_YOUR_HOST_AKA_LOCAL_MACHINE
 #AWS_SECRET_ACCESS_KEY=THIS_VALUE_SHOULD_HAVE_BEEN_PASSED_FROM_YOUR_HOST_AKA_LOCAL_MACHINE
 
-BZ_BASE_URL=http://canvasweb
+BZ_BASE_URL=http://platformweb
```
Comment (sadleb) [inline docker-compose/.env:15]: I don’t understand this. Seems like this pile lead to all sorts of problems.

Comment (hcli-work) [reply docker-compose/.env:15]: I know!!
This is how it's configured in Heroku though: 
image

I'm frightened

Comment (hcli-work) [reply docker-compose/.env:15]: image

Comment (hcli-work) [reply docker-compose/.env:15]: that's from our Heroku settings for portal-bebraven-dot-org

**Review by sadleb — APPROVED** [review]
Lgtm

**Pull request merged**